### PR TITLE
Backed enum value changed to Atomic instead of scalar int or strings

### LINKED ISSUE #9938: Enum values as class strings
Basically here's working code sample:
https://psalm.dev/r/c71751d4fe

Also is it possible to create a hint for enum values to be all class-strings?:

### PR DESCRIPTION
**Pull request by tuqqu**
Closes #9938

### CODE REVIEW

**Comment by weirdan** [comment]
Since there are no BC breaks, you can target `5.x` instead of master.

**Comment by tuqqu** [comment]
@weirdan the target branch is changed

**Comment by weirdan** [comment]
>  the target branch is changed

On second thought, the following would break:

```php
fn(EnumCaseStorage $s): int|string|null { return $s->value; }
```

It looks like I relied on automated checks too much. I've changed the target branch back to `master`, sorry about this confusion.

File: src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/AtomicPropertyFetchAnalyzer.php [hunk 1]
```diff
@@ -1034,10 +1033,9 @@ private static function handleEnumValue(
         $case_values = [];
 
         foreach ($enum_cases as $enum_case) {
```
Comment (weirdan) [inline src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/AtomicPropertyFetchAnalyzer.php:1035]: This loop body essentially becomes:

```php
foreach ($enum_cases as $enum_case) {
    $case_values[] = $enum_case->value ?? new TMixed();
}
```

File: src/Psalm/Internal/Provider/ReturnTypeProvider/GetObjectVarsReturnTypeProvider.php [hunk 2]
```diff
@@ -63,11 +63,12 @@ public static function getGetObjectVarsReturnType(
                     return new TKeyedArray($properties);
                 }
                 $enum_case_storage = $enum_classlike_storage->enum_cases[$object_type->case_name];
-                if (is_int($enum_case_storage->value)) {
-                    $properties['value'] = new Union([new Atomic\TLiteralInt($enum_case_storage->value)]);
-                } elseif (is_string($enum_case_storage->value)) {
-                    $properties['value'] = new Union([Type::getAtomicStringFromLiteral($enum_case_storage->value)]);
+
+                if ($enum_case_storage->value instanceof TLiteralString
+                    || $enum_case_storage->value instanceof TLiteralInt) {
```
Comment (weirdan) [inline src/Psalm/Internal/Provider/ReturnTypeProvider/GetObjectVarsReturnTypeProvider.php:68]:
```suggestion
                if ($enum_case_storage->value !== null) {
```

**Comment by weirdan** [comment]
Thanks!

**Pull request merged**